### ISSUE TITLE
[Toolbar][Accessibility] VoiceOver does not announce search results

### ISSUE THREAD

**Issue by natastro** [mozilla-mobile/firefox-ios#28060]
## Prerequisites:
* Turn on VoiceOver
* Turn on 'Homepage Search Bar' located in the Feature Flag menu
* Found in Firefox Beta 141.0 on iOS

## STR:
1. Turn on VoiceOver
1. Open Firefox Beta app
1. Navigate to the 'Search or enter address button'. Double tap to activate
1. A text input with a keyboard should appear on screen with focus starting in the text input. Begin to type a search query starting with the first letter
1. After you've typed the first letter, observe VoiceOver output

## Actual: 
In the video below, when typing the letter 'f' and then 'o', VoiceOver only reads autocomplete suggestions. Visually, there are several results that appear on screen - from Google Search and Firefox Suggest. None of these are read aloud

## Expected:
Once VoiceOver has finished its initial announcement, there should be a 2 to 3 second pause followed by announcing that there are a number of results available. The announcement should also specify that results are 'above' - if a VoiceOver user begins to swipe right they're taken to the Cancel button

## Device & build:
* Device: iPhone 15, iOS 18.5
* Build version: Firefox Beta 141.0
* First seen version: First time I've seen this issue

https://github.com/user-attachments/assets/1759a625-48fe-41cb-988c-186b07878249

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-12871)
